raft/src/main/java/org/apache/kafka/raft/internals/BatchMemoryPool.java

```diff
@@ -72,7 +74,13 @@ public void release(ByteBuffer previouslyAllocated) {
                     + previouslyAllocated.limit());
             }
 
-            free.offer(previouslyAllocated);
+            // Free the buffer if the number of pooled buffers is already the maximum number of batches.
+            // Otherwise return the buffer to the memory pool.
+            if (free.size() >= maxBatches) {
```

Comment:
Perhaps we should rename `maxBatches` since it is no longer serving as a max. How about `maxRetainedBatches` or something like that since it is still a bound on the number of batches which the pool can hold onto indefinitely.

Reply:
Yes. Fixed the variable name.